### PR TITLE
Bonded token amount with timestamp.

### LINKED ISSUE #2442: Bonded token amount with timestamp.
We need to track bonded token of users in system so we need to create a `bonded_token_track` table to insert a new row when some account delegate or unbond completed and save with timestamp(block).

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed: #2442

## Implementation details
Impelment table that store bonded token amount with timestamp.

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: flusher/flusher/db.py [hunk 1]
```diff
@@ -319,3 +320,11 @@ def Column(*args, **kwargs):
     Column("answer", CustomVoteOption),
     Column("tx_id", sa.Integer, sa.ForeignKey("transactions.id")),
 )
+
+shares_tracks = sa.Table(
```
Comment (taobun) [inline flusher/flusher/db.py]: historical_bonded_token_on_validators.

File: flusher/flusher/db.py [hunk 2]
```diff
@@ -319,3 +320,11 @@ def Column(*args, **kwargs):
...
+    "shares_tracks",
+    metadata,
+    Column("validator_id", sa.Integer, sa.ForeignKey("validators.id"), primary_key=True),
+    Column("shares_amount", sa.DECIMAL),
```
Comment (taobun) [inline flusher/flusher/db.py]: bonded_tokens

File: flusher/flusher/handler.py [hunk 3]
```diff
@@ -135,13 +136,28 @@ def handle_set_validator(self, msg):
...
+            }
+        )
 
     def handle_update_validator(self, msg):
         self.conn.execute(
             validators.update()
             .where(validators.c.operator_address == msg["operator_address"])
             .values(**msg)
         )
+        if "delegator_shares" in msg.keys():
```
Comment (taobun) [inline flusher/flusher/handler.py]: tokens

File: flusher/flusher/db.py [hunk 4]
```diff
@@ -234,6 +234,7 @@ def Column(*args, **kwargs):
     Column("status", sa.Boolean, default=False),
     Column("status_since", CustomDateTime, default=0),
     Column("accumulated_commission", sa.String),  # uband suffix
+    Column("last_update", CustomDateTime),
```
Comment (taobun) [inline flusher/flusher/db.py]: Remove it for now

**Review by taobun — CHANGES_REQUESTED** [review]

File: flusher/flusher/handler.py [hunk 5]
```diff
@@ -126,17 +127,30 @@ def handle_new_raw_report(self, msg):
...
         msg["account_id"] = self.get_account_id(msg["delegator_address"])
         del msg["delegator_address"]
-        if self.get_validator_id(msg["operator_address"]) is None:
-            self.conn.execute(validators.insert(), msg)
-        else:
-            condition = True
-            for col in validators.primary_key.columns.values():
-                condition = (col == msg[col.name]) & condition
-            self.conn.execute(validators.update().where(condition).values(**msg))
+        self.conn.execute(validators.insert(), msg)
```
Comment (taobun) [inline flusher/flusher/handler.py]: Why we removed the validator exited before update?

**Review by taobun — CHANGES_REQUESTED** [review]

File: flusher/flusher/handler.py [hunk 6]
```diff
@@ -126,17 +127,30 @@ def handle_new_raw_report(self, msg):
...
+        self.handle_new_historical_bonded_token_on_validator(
+            {
+                "validator_id": self.get_validator_id(msg["operator_address"]),
+                "bonded_tokens": msg["tokens"],
+                "timestamp": last_update,
+            }
+        )
 
     def handle_update_validator(self, msg):
+        if "tokens" in msg.keys():
```
Comment (sorawit) [inline flusher/flusher/handler.py]: don't think `.keys()` is necessary

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**